slippi-stats.js

```diff
@@ -71,250 +76,311 @@ var final_player_name = user_player
 
 console.log(`${files.length} replays found.`)
 
-for (i = 0; i < files.length; i++) {
-    try {
-        const game = new SlippiGame(files[i])
-        const settings = game.getSettings()
-        const metadata = game.getMetadata()
-        try {
-            game_seconds = Math.floor(metadata.lastFrame / 60)
-            game_length = Math.floor(game_seconds / 60) + ":" + (game_seconds % 60 ? (game_seconds % 60).toString().padStart(2, '0') : '00')
-            total_seconds += game_seconds
-        }
-        catch(err) {
-            console.log(`${i}: Error reading replay metadata (${files[i]}). Ignoring results...`)
-            continue
-        }
-        if (settings.players.length !== 2) {
-            console.log(`${i}: More than 2 players (${files[i]}). Ignoring results...`)
-            continue
-        }
+const staticPool = new StaticPool({
+    size: cores,
+    task: (args) => {
+        // everything in this closure is isolated from the rest of the script.
+        // if you need variables in here, you have to pass them through the workerData, aka "args".
+        // if you need to mutate data from the outside, you need to return that data on the "data"
+        // variable and process it later as part of "results".
```

Comment:
This comment explains how to provide data to the thread and how to return data from it. Figured it was important enough to leave in the code itself.